contrib/epee/include/net/http_client.h

```diff
@@ -385,11 +385,9 @@ using namespace std;
 
 					req_buff += "\r\n";
 					//--
-
-					bool res = m_net_client.send(req_buff, timeout);
-					CHECK_AND_ASSERT_MES(res, false, "HTTP_CLIENT: Failed to SEND");
 					if(body.size())
-						res = m_net_client.send(body, timeout);
+						req_buff += body;
+					bool res = m_net_client.send(req_buff, timeout);
```

Comment:
The writes were most likely split to avoid the additional data copy. ASIO does allow scatter-gather writes, but the internal `send` function would have to be updated to support that feature.